target/scripts/startup/setup.d/dmarc_dkim_spf.sh

```diff
@@ -15,11 +15,8 @@ function _setup_opendkim() {
 
     _log 'trace' "Adding OpenDKIM to Postfix's milters"
     postconf 'dkim_milter = inet:localhost:8891'
-    # shellcheck disable=SC2016
-    sed -i -E                                            \
-      -e '/\$dkim_milter/! s|^(smtpd_milters =.*)|\1 \$dkim_milter|g'     \
-      -e '/\$dkim_milter/! s|^(non_smtpd_milters =.*)|\1 \$dkim_milter|g' \
-      /etc/postfix/main.cf
+    _append_to_option_in_postfix_main 'smtpd_milters' "\$dkim_milter"
```

Comment:
```suggestion
    _append_to_option_in_postfix_main 'smtpd_milters' '$dkim_milter'
```

Reply:
I think the shellcheck lint will complain (_or did with `grep` unless I did `"\$value"` instead of `'$value'`_), and may affect `sed`? I recall some similar problem when passing strings like this to a function that'd use `sed` with a variable reference...

@georglauterbach has checked that this is working as intended with repeat runs? I don't think we have any specific test for it. My earlier advice on the original approach to this was to just ensure we reset the config files via a copy at startup, avoiding the issue (_so long as the change is not applied again via a `setup` command or triggered by `check-for-changes.sh`_).

Reply:
@casperklein  The quoting is done in order to not use `# shellcheck disable=SC...` above.

@polarathene I can double-check, but it may also be a good idea to ask @wligtenberg whether #3380 was properly solved for him. If so, this would indicate success. I would check nevertheless, but don't know when exactly.

Reply:
See https://github.com/docker-mailserver/docker-mailserver/pull/3484#issuecomment-1684998323